### PR TITLE
add dex operator alerts in working hours

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards: https://github.com/giantswarm/roadmap/issues/1809

This PR adds alerts for dex-operator in business hours.
`DexSecretExpired` pages in case a dex oauth secret was not renewed by dex-operator.
`ManagementClusterDexAppMissing` pages in case there is no registered dex oauth app for a management cluster reconciled by dex-operator.

### Checklist

- [ ] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/dex.rules.yml [hunk 1]
```diff
@@ -22,3 +22,27 @@ spec:
         severity: page
         team: rainbow
         topic: dex
+    - alert: DexSecretExpired
+      annotations:
+        description: '{{`Dex operator running on {{ $labels.installation }} failed to renew secret of {{ $labels.app_registration_name }} for {{ $labels.app_owner }} on provider {{ $labels.provider_type }}.`}}'
```
Comment (marians) [inline helm/prometheus-rules/templates/alerting-rules/dex.rules.yml:27]: I think we should use the same name for the component everywhere.

```suggestion
        description: '{{`dex-operator running on {{ $labels.installation }} failed to renew secret of {{ $labels.app_registration_name }} for {{ $labels.app_owner }} on provider {{ $labels.provider_type }}.`}}'
```

File: helm/prometheus-rules/templates/alerting-rules/dex.rules.yml [hunk 2]
```diff
@@ -22,3 +22,27 @@ spec:
...
+      for: 30m
+      labels:
+        area: kaas
+        cancel_if_outside_working_hours: "true"
+        severity: page
+        team: rainbow
+        topic: dex
+    - alert: ManagementClusterDexAppMissing
+      annotations:
+        description: '{{`Dex operator did not register a dex-app in {{ $labels.app_namespace }} namespace on {{ $labels.installation }}.`}}'
```
Comment (marians) [inline helm/prometheus-rules/templates/alerting-rules/dex.rules.yml:39]:
```suggestion
        description: '{{`dex-operator did not register a dex-app in {{ $labels.app_namespace }} namespace on {{ $labels.installation }}.`}}'
```

Comment (mcbenjemaa) [inline helm/prometheus-rules/templates/alerting-rules/dex.rules.yml:39]: There's no `installation` label.
I prefer to not specify it. as it's generated in the Opsgenie alert anyway.

Comment (anvddriesch) [reply helm/prometheus-rules/templates/alerting-rules/dex.rules.yml]: makes sense!

**Review by marians — APPROVED** [review]

**Review by mcbenjemaa — APPROVED** [review]

**Pull request merged**